readthedocs/api/v3/views.py

```diff
@@ -126,6 +129,20 @@ class ProjectsViewSet(APIv3Settings, NestedViewSetMixin, ProjectQuerySetMixin,
         'active_versions.last_build.config',
     ]
 
+    def get_serializer_class(self):
+        """
+        Return correct serializer depending on the action.
+
+        For GET it returns a serializer with many fields and on PUT/PATCH/POST,
+        it return a serializer to validate just a few fields.
```

Comment:
Does this not allow us to update the entire project via API?

Reply:
No, in theory only the fields from `ProjectCreateSerializer.Meta.fields` are allowed (name, language, repository and project_url). If there are extra fields passed, they are ignored.

We may want to add more, though.

Reply:
Added.